src/org/zaproxy/zap/extension/ascanrulesBeta/BufferOverflow.java

```diff
@@ -0,0 +1,180 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Buffer Overflow an active scan rule
+ * Copyright (C) 2015 Institute for Defense Analyses
+ * @author Mark Rader based upon the example active scanner by psiinon
+ * 
+ */
+
+package org.zaproxy.zap.extension.ascanrulesBeta;
+
+
+
+
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+
+public class BufferOverflow extends AbstractAppParamPlugin  {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanbeta.bufferoverflow.";
+	private static final int PLUGIN_ID = 30001;
+	private static Logger log = Logger.getLogger(BufferOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		try {
+			// This is where you change the 'good' request to attack the application
+			// You can make multiple requests if needed
+			String checkStringHeader1 = "Connection: close";  // Un natural close
+			String errorBufferOverflowMessage = "Potential Buffer Overflow.  The script closed the connection and threw a 500 Internal Server Error";
+			// Always use getNewMsg() for each new request
+			msg = getNewMsg();

```

Comment:
It's not necessary to create a new message (the one passed as parameter is already a new message).